packages/core/src/browser/label-provider.ts

```diff
@@ -47,10 +47,21 @@ export const FOLDER_ICON = DEFAULT_FOLDER_ICON;
 export const FILE_ICON = DEFAULT_FILE_ICON;
 
 export const LabelProviderContribution = Symbol('LabelProviderContribution');
+/**
+ * A {@link LabelProviderContribution} determines how specific elements/nodes are displayed in the workbench.
+ * Theia views use a common {@link LabelProvider} to determine the label and/or an icon for elements shown in the UI. This includes elements in lists
+ * and trees, but also view specific locations like headers. The common {@link LabelProvider} collects all {@links LabelProviderContribution} and delegates
+ * to the contribution with the highest priority. This is determined via calling the {@link LabelProviderContribution.canHandle} function, so contributions
+ * define which elements they are responsible for.
+ * As an example, a {@links LabelProviderContribution} can contribute an icon for a specific file type. This would be used in multiple locations displaying files,
+ * e.g. the file explorer, the header of editors or in file dialogues. Therefore, as arbitrary views can consume LabelProviderContributions, they must be generic
```

Comment:
> this would be used in multiple locations displaying files,  e.g. the file explorer, the header of editors or in file dialogues

I believe this is redundant, we already mention earlier where the label-provider can be used.

Reply:
I agree that the information is strictly speaking redudant. However, I think adding an example helps users to understand the purpose and most importantly that LabelProviderContributions must be generic. I personally think that understandability is more important compared to "no redudancy" when it comes to documentation. Please be aware that concepts such as the Label Provider and its contributions are not super trivial to understand for developers that have not used platforms such as Theia or Eclipse before.
So I would personally prefer to keep the section, if you do not aproove it, I will remove it though.

Reply:
> I agree that the information is strictly speaking redudant. However, I think adding an example helps users to understand the purpose and most importantly that LabelProviderContributions must be generic. 

@JonasHelming I mean that in the same part of the documentation, you already mention where it can be used:
https://github.com/eclipse-theia/theia/pull/8569/files#diff-339f915501a5afbb7359ed618ec424dfR52-R53, and then a little later mention it again. You can add actual examples earlier if you'd like but I'd rather the documentation not repeat itself.

Reply:
Yes, in the first part, I mention generically, where it can be used. At the beginning of the part you find redundant, I introduce the example, because it helps to understand the sentence behind it. If you have the example of a file in mind, you understand the different places where it is shown and therefore, that the label provider for it must be generic. So I found the example more useful at this place. I had it in the first paragraph in a previous version, but then the intro becomes very long.
Anyways, long story short, I removed the example as you suggested.